client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/ReplaceIfUnmodifiedOperation.java

```diff
@@ -35,6 +36,7 @@ protected VersionedOperationResponse executeOperation(Transport transport) {
       transport.writeArray(key);
       transport.writeVInt(lifespan);
       transport.writeVInt(maxIdle);
+      codec.writeExpirationNanoTimes(transport, lifespanNanos, maxIdleNanos, internalFlags);
```

Comment:
IOW, pass in a new flag within the protocol itself (change ProtocolFlag and `org.infinispan.client.hotrod.Flag` classes) to add `NANOS_EXPIRATION_UNIT`, and then `transport.writeVInt(lifespan);` and `transport.writeVInt(maxIdle);` write nano values instead of milliseconds.